diff --git a/Distribution/PackageDescription/TH.hs b/Distribution/PackageDescription/TH.hs
index 3b5789b..8e735c4 100644
--- a/Distribution/PackageDescription/TH.hs
+++ b/Distribution/PackageDescription/TH.hs
@@ -23,7 +23,7 @@ import Distribution.Text
 import Distribution.Compat.ReadP
 import Distribution.Verbosity (silent)
 import Text.PrettyPrint
-import Distribution.PackageDescription.Parse (readPackageDescription)
+import Distribution.PackageDescription.Parsec (readGenericPackageDescription)
 import System.Directory (getCurrentDirectory, getDirectoryContents)
 import Data.List (isSuffixOf)
 import Language.Haskell.TH (Q, Exp, stringE, runIO)
@@ -49,7 +49,7 @@ packageVariable = renderField currentPackageDescription
 -- | Renders the package variable specified by the function, from a cabal file
 -- and the given path.
 packageVariableFrom :: Text a => FilePath -> (PackageDescription -> a) -> Q Exp
-packageVariableFrom s = renderField $ fmap packageDescription (readPackageDescription silent s)
+packageVariableFrom s = renderField $ fmap packageDescription (readGenericPackageDescription silent s)
 
 ------
 renderField :: Text b => IO a -> (a -> b) -> Q Exp
@@ -63,7 +63,7 @@ currentPackageDescription = fmap packageDescription $ do
   dir <- getCurrentDirectory
   cs <- cabalFiles dir
   case cs of
-    (c:_) -> readPackageDescription silent c
+    (c:_) -> readGenericPackageDescription silent c
     [] -> error $ "Couldn't find a cabal file in the current working directory (" ++ dir ++ ")"
 
 cabalFiles :: FilePath -> IO [FilePath]
